components/codec/src/lib.rs

```diff
@@ -3,6 +3,7 @@
 #![cfg_attr(test, feature(test))]
 #![feature(core_intrinsics)]
 #![feature(ptr_offset_from)]
+#![feature(specialization)]
```

Comment:
What are you using specialization for? It makes me nervous because there are some soundness issues and it is a bit buggy.

Reply:
It is for https://sourcegraph.com/github.com/tikv/tikv/-/compare/7893f22555c28e1d381a83782700d0c5e9404c6d...57a7f0f690efd585413daf2de72ad4823a526857#diff-818b1ebd037db1159c8f98da6ea1ea77R60 so that `?` will work as normal.

Reply:
ah, that is unfortunate. I don't know a better way, but I wish that there were.